### ISSUE TITLE
v2 Installation Process Changes

### ISSUE THREAD

**Issue by ecowden** [oracle/node-oracledb#724]
I've been trying out version 2 and it no longer works with our standard instantclient setup. Specifically, I _think_ the issues are:

* For development, on MacOS, we've used the standard `/opt/instantclient` folder. Version 2 seems to require instantclient to be in the `/lib` folder. Is this right?
* For CI, staging, and deployment, on Linux, we don't have as much control over file locations. We've worked around it through liberal use of `OCI_LIB_DIR` and `OCI_INC_DIR`. Do these no longer work? I've only done a quick smoke test, so there may be another issue I'm missing.

Is there any way to make the old installation process work with the new version?

**Comment by anthony-tuininga** [comment]
The environment variables OCI_LIB_DIR and OCI_INC_DIR are no longer relevant as documented in Chris' [post](https://blogs.oracle.com/opal/node-oracledb-2013-development-is-now-on-github) and the [release notes](https://github.com/oracle/node-oracledb/blob/dev-2.0/CHANGELOG.md). During building, OCI header and library files are not required, so there is no need to say where they are located!

ODPI-C uses runtime dynamic loading. This has the advantage of allowing one binary to handle 11.2, 12.1 and 12.2 Oracle client libraries without requiring recompilation. It does mean, however, that optimisations like the use of RPATH are no longer possible (for a number of technical reasons which I won't get into here). You must instead use environment variables like LD_LIBRARY_PATH or create a configuration file in /etc/ld.so.conf.d or create a symlink (on macOS) in ~/lib. See the ODPI-C [installation notes](https://oracle.github.io/odpi/doc/installation.html) for additional information.

In short, there is no way to make the old installation process work with the new version. We believe the new process is better, even if it is different -- but we are also looking at further improvements so any feedback is appreciated!

Can you provide the errors you are receiving? Or was this simply intended as feedback?

**Comment by cjbj** [comment]
To complete the link fest: here is the node-oracledb [2.0.13-dev installation doc](https://github.com/oracle/node-oracledb/blob/dev-2.0/INSTALL.md).  Comments etc welcome.

On macOS, this is all I have in `~/lib`:
```
$ ls -l ~/lib
ls -l ~/lib
total 8
lrwxr-xr-x  1 cjones  staff  48 20 Mar 15:12 libclntsh.dylib -> /Users/cjones/instantclient/libclntsh.dylib.12.1
```

The same should also be possible with `/usr/local/lib`.

**Comment by ecowden** [comment]
Thanks for the info!

> Can you provide the errors you are receiving? Or was this simply intended as feedback?

I suppose I'd consider it feedback at this point. Getting everything working and automated with the first version -- between workstations, continuous integration, and deployment -- was a significant effort. We'd _love_ to not have to repeat that process if possible.

Feedback provided, feel free to close this issue.

Thanks!

**Comment by anthony-tuininga** [comment]
You're welcome. Please provide additional feedback as it comes to you. I'll close this for now.
